### PR TITLE
Measure cycles instead of time for benchmarking

### PR DESCRIPTION
**Pull request by oflatt**
This PR does several things (sorry):
- changes the runtime to expose functions that measure cycles on arm and x86
- changes eggcc to use the updated bril fork that instruments the llvm code to use these
- changes all the benchmarks to not have loops that run them multiple times
- changes the benchmarks to have a single print statement before existing to remove I/O overhead
- copies polybench benchmarks into test folder so that we still test the printed matrices
- fixes parallelism in nightly so that it exits on first error

### CODE REVIEW

File: benchmarks/passing/bril/core/euclid.bril [hunk 1]
```diff
@@ -1,30 +1,6 @@
-# ARGS: 2550
+# ARGS: 38432 25
```
Comment (yihozhang) [inline benchmarks/passing/bril/core/euclid.bril:1]: 38432 and 25 are coprime so this benchmark is now too easy?

File: benchmarks/passing/bril/core/fact.bril [hunk 2]
```diff
@@ -1,27 +1,10 @@
-# ARGS: 9500
```
Comment (yihozhang) [inline benchmarks/passing/bril/core/fact.bril:1]: why's the input smaller now?

File: benchmarks/passing/bril/core/gcd.bril [hunk 3]
```diff
@@ -1,36 +1,13 @@
-# ARGS: 1190
+# ARGS: 1190 83
```
Comment (yihozhang) [inline benchmarks/passing/bril/core/gcd.bril:1]: 1190 and 83 are coprimes.

File: benchmarks/passing/bril/core/euclid.bril (hunk 1 above)
Comment (yihozhang) [reply benchmarks/passing/bril/core/euclid.bril:1]: Overmind, coprime actually takes longer for gcd to compute

**Review by yihozhang — APPROVED** [review]
lgtm

File: benchmarks/passing/bril/core/fact.bril (hunk 2 above)
Comment (oflatt) [reply benchmarks/passing/bril/core/fact.bril:1]: It was overflowing or something, but it didn't need to be that big.

**Pull request merged**